src/shared/components/section.tsx

```diff
@@ -23,7 +23,10 @@ export const Section: React.FC<IProps> = ({ section, dataSchema, formUiSchema, f
     }
   }
 
-  const onSubmit = (event: ISubmitEvent<IExperimentData>) => onDataChange && onDataChange(event.formData);
+  const onChange = (event: IChangeEvent<IExperimentData>) => {
+    // Immediately save the data.
+    onDataChange && onDataChange(event.formData);
```

Comment:
Should we standardize on `onDataChange?.(event.formData)`?

Reply:
The `?.` is a little weird, I guess they had to do that for parsing reasons.

Reply:
Standardize in what sense? Both `Experiment` and `Section` have the same API, but I guess you meant something else. Also, not sure about `?.` - where is that? ;)

Reply:
I just meant since we are using the existential operator already in other places in the code we should also use it here.

Instead of writing `onDataChange && onDataChange(event.formData)` you can write `onDataChange?.(event.formData)`.

Reply:
Ah, ok. Right, I need to get used to that. I think I use `?` and `!` mostly when TS is complaining about possible `undefined` value when it shouldn't (as I assign value a few lines above) to avoid silly `if`s. But `?` could be quite useful anyway and probably it won't go away, so we can use it.

Reply:
OK, now I also see what you mean with `?.`. Yeah, this dot is quite weird. Maybe let's use that for properties when it's hidden, but `callback?.(argument1, argument2)` might be pretty confusing for other devs.

Reply:
It's standard JavaScript now, not just a TypeScript invention, so I think expecting developers to get used to it is appropriate. From a making sense of it standpoint, there isn't a separate `?` operator, there's just the optional chaining operator `?.` which is used for both object dereferences and function calls, hence the need for the otherwise unexpected `.` when using optional chaining with a function call.